sql/opt/exec/execbuilder: TestExecBuild failed

sql/opt/exec/execbuilder.TestExecBuild [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3580398&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3580398&tab=artifacts#/) on master @ [2f2c517d2f2fd046ee25ad4dfe62c32f7eda6ac5](https://github.com/cockroachdb/cockroach/commits/2f2c517d2f2fd046ee25ad4dfe62c32f7eda6ac5):


```
                      estimated row count: 100 - 1,001 (100% of the table; stats collected <hidden> ago)
                      table: a@a_i_j_idx
                      spans: FULL SCAN
        but found (query options: "") :
            planning time: 10µs
            execution time: 100µs
            distribution: <hidden>
            vectorized: <hidden>
            rows read from KV: 176 (1.4 KiB)
            maximum memory usage: <hidden>
            network usage: <hidden>
            regions: <hidden>
            ·
            • limit
            │ nodes: <hidden>
            │ regions: <hidden>
            │ actual row count: 5
            │ KV time: 0µs
            │ KV contention time: 0µs
            │ KV rows read: 76
            │ KV bytes read: 608 B
            │ count: 5
            │
            └── • lookup join
                │ nodes: <hidden>
                │ regions: <hidden>
                │ actual row count: 5
                │ KV time: 0µs
                │ KV contention time: 0µs
                │ KV rows read: 76
                │ KV bytes read: 608 B
                │ table: b@b_k_s_idx
                │ equality: (j) = (k)
                │
                └── • scan
                      nodes: <hidden>
                      regions: <hidden>
                      actual row count: 100
                      KV time: 0µs
                      KV contention time: 0µs
                      KV rows read: 100
                      KV bytes read: 800 B
                      estimated row count: 100 - 1,001 (100% of the table; stats collected <hidden> ago)
                      table: a@a_i_j_idx
                      spans: FULL SCAN
--- done: testdata/limit with config local: 28 tests, 1 failures
--- done: testdata/distinct_on with config local: 32 tests, 0 failures
        --- FAIL: TestExecBuild/local/limit (10.50s)
=== RUN   TestExecBuild/local
    --- FAIL: TestExecBuild/local (0.00s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestExecBuild PKG=./pkg/sql/opt/exec/execbuilder TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
/cc @cockroachdb/sql-queries RaduBerinde
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/opt/exec/execbuilder.TestExecBuild [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3582804&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3582804&tab=artifacts#/) on master @ [96bb0bc7fd2ba18555c34aa9766d84903c6fd311](https://github.com/cockroachdb/cockroach/commits/96bb0bc7fd2ba18555c34aa9766d84903c6fd311):


```
                      table: a@a_i_j_idx
                      spans: FULL SCAN
        but found (query options: "") :
            planning time: 10µs
            execution time: 100µs
            distribution: <hidden>
            vectorized: <hidden>
            rows read from KV: 160 (1.3 KiB)
            maximum memory usage: <hidden>
            network usage: <hidden>
            regions: <hidden>
            ·
            • limit
            │ nodes: <hidden>
            │ regions: <hidden>
            │ actual row count: 5
            │ KV time: 0µs
            │ KV contention time: 0µs
            │ KV rows read: 76
            │ KV bytes read: 608 B
            │ estimated row count: 5
            │ count: 5
            │
            └── • lookup join
                │ nodes: <hidden>
                │ regions: <hidden>
                │ actual row count: 5
                │ KV time: 0µs
                │ KV contention time: 0µs
                │ KV rows read: 76
                │ KV bytes read: 608 B
                │ estimated row count: 100
                │ table: b@b_k_s_idx
                │ equality: (j) = (k)
                │
                └── • scan
                      nodes: <hidden>
                      regions: <hidden>
                      actual row count: 84
                      KV time: 0µs
                      KV contention time: 0µs
                      KV rows read: 84
                      KV bytes read: 672 B
                      estimated row count: 100 - 1,001 (100% of the table; stats collected <hidden> ago)
                      table: a@a_i_j_idx
                      spans: FULL SCAN
--- done: testdata/limit with config local: 28 tests, 1 failures
        --- FAIL: TestExecBuild/local/limit (4.00s)
=== RUN   TestExecBuild/local
    --- FAIL: TestExecBuild/local (0.00s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestExecBuild PKG=./pkg/sql/opt/exec/execbuilder TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
/cc @cockroachdb/sql-queries RaduBerinde
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/opt/exec/execbuilder/testdata/limit
+++ b/pkg/sql/opt/exec/execbuilder/testdata/limit
@@ -511,7 +511,7 @@ ANALYZE a;
 statement ok
 ANALYZE b;
 
-query T
+query T retry
 EXPLAIN ANALYZE SELECT * FROM a INNER LOOKUP JOIN b ON k = j ORDER BY i LIMIT 5;
 ----
 planning time: 10µs

